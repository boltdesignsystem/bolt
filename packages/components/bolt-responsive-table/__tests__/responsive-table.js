// Refer to https://github.com/boltdesignsystem/bolt/wiki/Jest-Test:-Example-Jest-Test for more testing examples
import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../table.schema';
// const { borderless, sticky_headers } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
});

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is a table',
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt Responsive Table component', () => {
  test('default', async () => {
    const results = await render(
      '@bolt-components-responsive-table/table.twig',
      {
        ...fixtures.defaultData,
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});
