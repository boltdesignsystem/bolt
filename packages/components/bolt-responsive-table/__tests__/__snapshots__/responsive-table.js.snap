// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Responsive Table component default 1`] = `
<div class="c-bolt-responsive-table js-bolt-responsive-table">
  <table>
  </table>
</div>
`;
